tests/test_pin.py

```diff
@@ -0,0 +1,65 @@
+from helpers import run_pipx_cli
+from package_info import PKG
+
+
+def test_pin(monkeypatch, capsys, pipx_temp_env, caplog):
+    assert not run_pipx_cli(["install", "pycowsay"])
+    assert not run_pipx_cli(["pin", "pycowsay"])
+    assert not run_pipx_cli(["upgrade", "pycowsay"])
+
+    assert "Not upgrading pinned package pycowsay" in caplog.text
+
+
+def test_pin_with_suffix(monkeypatch, capsys, pipx_temp_env, caplog):
+    assert not run_pipx_cli(["install", PKG["black"]["spec"], "--suffix", "@1"])
+    assert not run_pipx_cli(["pin", "black@1"])
+    assert not run_pipx_cli(["upgrade", "black@1"])
+
+    assert "Not upgrading pinned package black@1" in caplog.text
+
+
+def test_pin_warning(monkeypatch, capsys, pipx_temp_env, caplog):
+    assert not run_pipx_cli(["install", PKG["nox"]["spec"]])
+    assert not run_pipx_cli(["pin", "nox"])
+    assert not run_pipx_cli(["pin", "nox"])
+
+    assert "Package nox already pinned 😴" in caplog.text
+
+
+def test_pin_not_installed_package(monkeypatch, capsys, pipx_temp_env):
```

Comment:
This doesn't seem to be used in any of the tests. Accidentally copied?

```suggestion
def test_pin_not_installed_package(capsys, pipx_temp_env):
```

Reply:
Yes 😅, thanks for reminding!